src/iter/step_by.rs

```diff
@@ -0,0 +1,131 @@
+use super::plumbing::*;
+use super::*;
+use crate::math::div_round_up;
+use std::iter;
+use std::usize;
+
+/// `StepBy` is an iterator that skip `n` element between each yield. where `n` is the given step
+/// This struct is created by the [`step_by()`] method on [`IndexedParallelIterator`]
+///
+/// [`step_by()`]: trait.IndexedParallelIterator.html#method.step_by
+/// [`IndexedParallelIterator`]: trait.IndexedParallelIterator.html
+#[cfg(step_by)]
+#[must_use = "iterator adaptors are lazy and do nothing unless consumed"]
+#[derive(Debug, Clone)]
+pub struct StepBy<I: IndexedParallelIterator> {
+    base: I,
+    step: usize,
+}
+
+#[cfg(step_by)]
+impl<I> StepBy<I>
+    where
+        I: IndexedParallelIterator,
```

Comment:
The indentation is odd here and elsewhere -- please run `cargo fmt` to clean it up.